libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -385,4 +365,19 @@ void ReferencesResolver::validateYulIdentifierName(yul::YulString _name, SourceL
 			_location,
 			"The identifier name \"" + _name.str() + "\" is reserved."
 		);
+
+	auto declarations = m_resolver.nameFromCurrentScope(_name.str());
+	if (!declarations.empty())
+	{
+		SecondarySourceLocation ssl;
+		for (auto const* decl: declarations)
+			ssl.append("The shadowed declaration is here:", decl->location());
```

Comment:
For
```solidity
contract C {
	function f() external {
		assembly {
			function g(gasleft) { gasleft := 2 }
		}
	}
}
```
The `decl->location` would be empty.

Should we do something about global functions?